初回版

yusukebkk opened the pull request:
初回提出分です。

yusukebkk committed:
初回版

manaberuit commented on index.html:
```diff
@@ -0,0 +1,16 @@
+<body>
+    <button>aaa</button>
+    <div>aaa
+        <div> aaaaaaa</div>
```
これは？？

yusukebkk replied:
最初に色々テストしたコードがそのまま残ってしまっていました。失礼いたしました。

manaberuit commented on js/main.js:
```diff
@@ -0,0 +1,185 @@
+
+    //クイズのメイン処理
+    const makeQuestion = async (questionData,count = 0,score = 0) => {
+        reset();
```
関数実行の前に一言コメントをいれておくと良いですね。

yusukebkk replied:
関数実行の前にコメントを追加しました。

manaberuit commented on js/main.js:
```diff
@@ -0,0 +1,185 @@
+    }
+
+    //クイズのメイン処理
+    const makeQuestion = async (questionData,count = 0,score = 0) => {
```
`,`の後には空白を入れましょう。

yusukebkk replied:
修正しました

manaberuit commented on js/main.js:
```diff
@@ -0,0 +1,185 @@
+        reset();
+        loading();
+        //questionDataがない場合にquestionDataを取得する
+        if (typeof questionData === 'undefined'){
```
```
if (!questionData)
```
で良いかなと思います。

このように記述すると、`questionData`にちゃんと値が入っていないときに実行されるようになります。

yusukebkk replied:
ありがとうございます。こっちの方がすっきりしますね。

manaberuit commented:
一つのファイルのコード量が多くなってしまっているので、例えば、関数の定義はfunctions.jsなどの他のファイルに分けましょう。

manaberuit commented on js/main.js:
```diff
@@ -0,0 +1,185 @@
+    //ボタンの追加処理
+    const addChoiceButtons = (correctAnswer,incorrectAnswers,score,count,questionData) =>{
+        //全ての選択肢のリスト作成
+        let buttonList =[]
```
constで良いのでは？

yusukebkk replied:
constでもリストの中身は変えられるんですね！constに変更しました。

manaberuit replied:
そうですね！！
ですので、配列やオブジェクトは基本constで宣言します！！

manaberuit commented on js/main.js:
```diff
@@ -0,0 +1,185 @@
+    const addChoiceButtons = (correctAnswer,incorrectAnswers,score,count,questionData) =>{
+        //全ての選択肢のリスト作成
+        let buttonList =[]
+        let allChoice = incorrectAnswers;
```
constで良いのでは？

yusukebkk replied:
constに変更しました。

manaberuit commented on js/main.js:
```diff
@@ -0,0 +1,185 @@
+    const addChoiceButtons = (correctAnswer,incorrectAnswers,score,count,questionData) =>{
+        //全ての選択肢のリスト作成
+        let buttonList =[]
+        let allChoice = incorrectAnswers;
```
また、配列の代入の場合、値渡しではなく、参照渡しとなりますが、その点は問題ないですか？
https://qiita.com/migi/items/3417c2de685c368faab1

yusukebkk replied:
すっかり忘れていました。最終的にallChoiceの中身を変更すると、incorrectAnswersの中身も変わっちゃうんですね。今回はincorrectAnswerをその後の処理で使わないためそのままにしてしまったのですが、concatもしくはfor文を使って書き換えた方が良いのでしょうか？

manaberuit replied:
その通りです！！
使わないとしても、参照渡しはバグの要因となるので、なるべく避けたいところです。
sliceという便利な組み込みメソッドがあるので、それで書き換えてみましょう。
https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Global_Objects/Array/slice

manaberuit commented on js/main.js:
```diff
@@ -0,0 +1,185 @@
+        } );
+    }
+    //ボタンの追加処理
+    const createButton = (text, func) => {
```
funcという変数名はよくないです。
この引数でどういった処理の関数を受け取るかが直感的に分かる変数名にしましょう。
例えば、`eventListener`でも良いと思います。

yusukebkk replied:
ありがとうございます。eventListnerに変更しました。

manaberuit commented on js/main.js:
```diff
@@ -0,0 +1,185 @@
+{
+    
+    //apiで値を取得する
+    const getQuestionData = async () => {
```
async/awaitをしっかり使えていて良いですね。

ただ、今回はクイズ取得の処理とローディングの処理を非同期で走らせたいので、そのように処理を変更してみましょう。

yusukebkk replied:
処理を変更しました。

manaberuit reviewed (commented):
いくつかコメントしましたので、ご確認ください。

yusukebkk committed:
HTML修正

htmlの不要箇所を削除しました。

yusukebkk committed:
コメントの追加とカンマ後のスペースの追加

yusukebkk committed:
クイズ取得の処理とローディングの処理を非同期で走らせるように変更

yusukebkk committed:
questionDataの有無の判定処理を変更

yusukebkk committed:
letになっていた変数をconstに変更

yusukebkk committed:
変数名をfuncからeventListnerに変更

変数名をfuncからeventListnerに変更
余計なconsole.logを削除しました

yusukebkk committed:
jsファイルを機能ごとに分割

-indexファイルのjsファイルの参照情報を変更
-function.jsに細かい関数を移動
-get-api-data.jsにapi取得処理を移動
-quiz-main-fumction.jsにクイズ関連のメイン処理を移動

yusukebkk commented:
いただいたフィードバックを元に修正を行いました。

manaberuit reviewed (approved):
OKです！！
以下だけ修正しておいてください。
https://github.com/yusukebkk/quizapp/pull/1#discussion_r295100172

その後はマージしてしまって構いませんが、不安でしたら最後レビュー依頼投げてください。

また、ファイルに関してはそこまで分割しなくても良いかな～とは思います。
（修正はしなくて良いです。）
main.jsとfunctions.jsのみで十分かなと。
ただ、フレームワークとか使いだすと、今回のような感じでファイルを分割していくことになるので、心がけとしてはよいと思います！！

yusukebkk committed:
sliceを使った処理に変更

yusukebkk commented:
最後の修正を行いました。
